Ais omar patch 1 (#13)
brown-ccv/honeycomb-docs: docs/version_control.md

| git push                      | upload staged commit to repo                                       |
| git pull                      | get remote repo commits and download \(try and resolve conflicts\) |
| git clone &lt;url&gt;         | download entire repository                                         |
| git branch &lt;new-branch&gt; | create a new branch on your currently checked out (HEAD) branch    |

> | git checkout &lt;branch&gt;   | checkout the branch you want to use                                | 

Reviewer: You can do those two steps in on `git checkout -b <branch-name> ` Maybe worth mentioning?

But also the instructions from @RobertGemmaJr  should match this. I think if we use `switch` and `checkout` for the same operation we may confuse them unless we explicitly mention that there are different commands that do the same thing

Reviewer 2: Yeah I can make my changes to `git checkout -b`! No problem

| git push                      | upload staged commit to repo                                       |
| git pull                      | get remote repo commits and download \(try and resolve conflicts\) |
| git clone &lt;url&gt;         | download entire repository                                         |
| git checkout &lt;branch&gt;   | checkout and create the branch you want to use                                | 
  
  
```bash
# clone project 
git clone https://github.com/brown-ccv/honeycomb.git
# create branch with your feature
git checkout -b feature_name
# check the status of your repositoey
git status
# commit file contents to the local repository
git commit -am "My feature is ready"
# specific message
# push file contents to the remote (i.e. cloud) repository
git push origin feature_name
```

### Comment styles

> We encourage using [Commitizen](http://commitizen.github.io/cz-cli/), a great tool for writing angular commits - this will create a standardized commit format which makes for easier change logging and more sane messages.

Reviewer 3: I'm not sure what "angular" means here. Might we want to make it a bit more targeted at people who haven't used git before?: 
```suggestion
We encourage using [Commitizen](http://commitizen.github.io/cz-cli/), a great tool for recording descriptions of commits in a standardized format which makes it easier for people to understand what changed in the code.
```

Reviewer: Angular commits are also known as [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) - we could use that notation instead and set the link

> ### Alternative options 

Reviewer 3: Might this section on the GUI be better put at the top? It's possibly much more intuitive for beginners. And should it be the main way we suggest people use the tools?

Reviewer: I'd be okay with that!

Reviewer 2: I definitely think the GUI solution is something we should share briefly during the workshop. I know I do almost all of my commit/push/pull from VS Code and only touch the command line if I'm doing some more tricky stuff.

Imo so long as people have a grasp of what they (commit/push/pull) and what's happening when they use the gui it's safe to use and easier for beginners

git checkout -b feature_name
# check the status of your repositoey
git status
# commit file contents to the local repository
git commit -am "My feature is ready"
# specific message
# push file contents to the remote (i.e. cloud) repository
git push origin feature_name
```
### Alternative options 
  
Instead of using commands in the terminal, you can also download [GitHub desktop](https://desktop.github.com/). It is very intuitive to use. 
  
#### Clone repository 

![](assets/git_clone.png)
  
#### Select a branch or create a new branch

![](assets/git_branch.png)
  
#### Commit changes and push

![](assets/git_commit.png)

## Create a Pull Request 

Pull requests are useful before you merge your branch with the main branch. You can request a review from your colleagues and check for any conflicts with the main branch. After you pushed all the changes to your branch, you can go to the original GitHub repository and click on the pull request.  

![](assets/pull_request.png)

![](assets/pull_request_info.png)

## Best Practices

 ### Git Workflow
  
 We recommend using a simple flow based on following rules:

- Use topic/feature branches, no direct commits on main.

> - Perform tests and code reviews before merges into main, not afterwards.

Reviewer 2: Is there an action for running the tests on push? Perhaps we could make one and instruct the users to set up that action on top of firebase

Reviewer 2: cc @eldu

> - Everyone starts from main, and targets main.

Reviewer 2: ```suggestion
- Every branch starts from main, and targets main.
```

- Commit messages reflect intent.

**Branches**

* `main` is the default branch and where releases are made off. This branch should be in clean/working conditions at all times. This branch is protected and can only be merged from Pull Requests for topic branches

> * `topic` branches are created for new features, fixes, or really any changes. E.g, `fix-task-trial2-stuck-button`

Reviewer 2: We should stick to either `-` or `*` for the bullet points, `-` seems to be much more common

### Comment styles

We encourage using [Commitizen](http://commitizen.github.io/cz-cli/), a great tool for writing angular commits - this will create a standardized commit format which makes for easier change logging and more sane messages.
  

> ### Alternative options 

Reviewer: I think this content is best before the ## Best Procatices section

Maybe in best practices we have an explicit explanation about workflow

I'll put a suggestion below

  
Instead of using commands in the terminal, you can also download [GitHub desktop](https://desktop.github.com/). It is very intuitive to use. 
  
#### Clone repository 

![](assets/git_clone.png)
  
#### Select a branch or create a new branch

![](assets/git_branch.png)
  
#### Commit changes and push

![](assets/git_commit.png)

>   

Reviewer: ```suggestion
 ### Git Workflow
  
 We recommend using a simple flow based on following rules:

- Use topic/feature branches, no direct commits on main.
- Perform tests and code reviews before merges into main, not afterwards.
- Everyone starts from main, and targets main.
- Commit messages reflect intent.

**Branches**

* `main` is the default branch and where releases are made off. This branch should be in clean/working conditions at all times. This branch is protected and can only be merged from Pull Requests for topic branches
* `topic` branches are created for new features, fixes, or really any changes. E.g, `fix-task-trial2-stuck-button`

This flow is sometimes referred to as [Feature Branch Workflow](https://docs.gitlab.com/ee/gitlab-basics/feature_branch_workflow.html)
```
